swiftlang/swift-docc
# Use only canonical paths for symbol breadcrumbs

Bug/issue #, if applicable: rdar://77922907

## Summary

This changes how DocC computes "breadcrumbs" for symbols to only use the canonical path and to use different paths for each language representation.
Before these changes, DocC would use the same "breadcrumbs" for each language representation, always based on the shortest path to the symbol, including curation. This meant, among other things, that:
- C API with Swift refinements wouldn't reflect the symbol hierarchy on the Swift version of the page (because the path to the C top-level function was always shorter)
- members that were organized outside of the canonical container location would no longer include a link to the container symbol on their page.

With these changes, symbol breadcrumbs ignore authored curation and only consider the canonical path for each language representation.  For example:

<img width="1164" alt="Screenshot 2024-11-01 at 11 25 33" src="https://github.com/user-attachments/assets/70e4567d-2d8f-401b-9338-ba7a3e01a15c">

Articles and tutorial content continue to base their breadcrumbs on authored curation. 

## Dependencies

None.

## Testing

- Build and preview documentation for the "GeometricalShapes" test catalog
  ```
  swift run docc preview Tests/SwiftDocCTests/Test\ Bundles/GeometricalShapes.docc 
  ```

- Click through the different pages and inspect the breadcrumbs for both language representations.
  - Note that `center` and `radius` are members in each language representation but that `intersects(_:)`, `isEmpty`, etc. are top-level functions in C/Objective-C.
  - Note that `init(center:radius:)` only exist in Swift and that `TLACircleMake` only exist in C/Objective-C.

<img width="916" alt="Screenshot 2024-11-01 at 11 39 49" src="https://github.com/user-attachments/assets/53eebeb8-ba8d-4f44-a40f-6f52d541ba07">

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
